competition/track2/train/train.py

```
@@ -129,8 +122,12 @@ def train(input_path, output_path):
                     dx = trans_next[0, 0] - trans_cur[0, 0]
                     dy = trans_next[1, 0] - trans_cur[1, 0]
                     dheading = next_heading - current_heading
-                    events = vehicle_data[float(sim_time)]["events"]
-                    if all(value == 0 for value in events.values()):
+                    events = vehicle_data[float(sim_time)].events
+                    attributes = [a for a in dir(events) if not a.startswith('__')]
+                    events_values = []
+                    for a in attributes:
+                        events_values.append(getattr(events, a))
+                    if all(value == 0 for value in events_values):
                         terminal = 0
                     else:
                         terminal = 1
```

@alfredgu001324 @cz0204 
1) Is this code correctly doing what it is expected to do?
2) Consider simplifying this code.

---

1. It is iterating through all the object's attributes and checking their values, since the current data format is no longer a dictionary so cannot check for dict.values()
2. Not too sure how to simplify this

---

@alfredgu001324 @cz0204 
1. Can you check and confirm whether the code is correctly performing its intended functionality?
2. A potential simplification is as follows:
```suggestion
                    events = vehicle_data[float(sim_time)].events
                    terminal = 0
                    for name, value in events._asdict().items():
                        if bool(value):
                            terminal = 1
                            break
```

---

@alfredgu001324 @cz0204 
1. Can you check and confirm whether the code is correctly performing its intended functionality?
2. When should the `terminal=1` and when should `terminal=0`?
3. Must there be at least one instance of `terminal=1` in `terminals` for the CQL training to work?
4. A potential simplification is as follows:
```suggestion
                    events = vehicle_data[float(sim_time)].events
                    terminal = 0
                    for name, value in events._asdict().items():
                        if bool(value):
                            terminal = 1
                            break
```

---

1. Yes
2 & 3. I am not too sure. Maybe @cz0204  will know this better
3. Does event object has _asdict()?

---

2. if anyone of the events is True, then terminal is 1, otherwise it is 0.
3. yes, terminal has to contain at least 1 in offline training to work. Otherwise we have to setup another feature in CQL, which is not used here in our dataset. So this feature is not included in current code.

---

@cz0204 @alfredgu001324 

1. The `events` contains `collision` which is a list. The equality check `all(value == 0 ... )` will always evaluate to `False` because comparing a list to an int gives false.
2. After making necessary corrections, could you rectify this CQL code such that it can be trained using the [example dataset](https://github.com/smarts-project/smarts-project.offline-datasets/blob/master/offline_dataset.rar)?

---

@cz0204 @Adaickalavan 
3. It seems that the current dummy data set only contains terminal 0 (total of 143 pieces of data), and thus reporting an 'index out of range' error